projects/evo-ui-kit/src/lib/modules/evo-paginator/evo-paginator.component.ts

```diff
@@ -0,0 +1,101 @@
+import { Component, Output, EventEmitter, Input, ChangeDetectionStrategy } from '@angular/core';
+
+export interface PageEvent {
+    currentPage: number;
+    previousPage: number;
+    pageSize: number;
+    pagesTotal: number;
+}
+
+// use only odd numbers please
+const PAGES_VISIBLE = 5;
+
+@Component({
+    selector: 'evo-paginator',
+    templateUrl: './evo-paginator.component.html',
+    styleUrls: ['./evo-paginator.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class EvoPaginatorComponent {
+    @Output() pageClick: EventEmitter<PageEvent> = new EventEmitter<PageEvent>();
```

Comment:
duck: Duck = new Duck()

Reply:
?

Reply:
Класическая шутка о статическом тайпинге. Не слишком ли много уток? 

Там где тип может быть легко выведен из определения, тип лучше опустить, так как меньше копипасты и когнитивной нагрузки при чтении кода. 

```suggestion
    @Output() pageClick = new EventEmitter<PageEvent>();
```

Reply:
Это больше про кодстайл, но да, согласен, упростил.